### ISSUE TITLE
Use localStorage

### ISSUE THREAD

**Issue by agentcooper** [agentcooper/typescript-play#1]
Code should be saved to localStorage when closing playground.

**Comment by karol-majewski** [comment]
@agentcooper What do you think about persisting compiler options as well?

**Comment by agentcooper** [comment]
@karol-majewski yes, this definitely should be done, however it becomes tricky with params coming from url (they probably should take precedence).

**Comment by Airblader** [comment]
I think the logic should go like this:

1. If there's any URL parameter set, use that and skip the other step. However, do update this into the localStorage now. 
2. Otherwise, restore from localStorage and update the URL to reflect the now loaded state. 

I wouldn't mix localStorage with URL parameters in cases where a single parameter isn't set. It's very unlikely that that's what a user would expect.
